src/modules/common/components/upload-file/index.tsx

```diff
@@ -0,0 +1,176 @@
+import { useState, useRef, useEffect } from "react";
+import { FiUpload, FiUploadCloud } from "react-icons/fi";
+
+type Props = {
+    separatedFiles?: boolean
+    required?: boolean
+    multiple?: boolean
+    name: string
+    preview?: boolean
+}
+
+export function UploadFile({ required, multiple, name, preview }: Props) {
```

Comment:
la funcion solo renderiza el componente, la logica se puede aislar luego en un proximo refactor
```suggestion
//TODO: Refactor this
export function UploadFile({ required, multiple, name, preview }: Props) {
```